src/stores/businessStore.ts

```diff
@@ -277,17 +282,24 @@ export const useBusinessStore = defineStore('business', {
       return new Promise((resolve, reject) => {
         const businessId = sessionStorage.getItem('BUSINESS_ID')
 
-        // if there is no business id, return error
+        // If there is no business id, return an error.
         if (!businessId) {
           reject(new Error('Missing business id'))
           return
         }
-
         LegalServices.fetchBusiness(businessId)
           .then(businessInfo => {
-            // set data to store
+            // Set data to store
             this.setBusinessInfo(businessInfo)
-            // return the business info object
+            if (this.isFirm) {
+              // Extract the operatingName from the businessInfo
+              const alternateName = businessInfo.alternateNames?.find(
+                x => x.identifier === businessInfo.identifier
+              )?.operatingName
+              // Set the operatingName if it exists
+              if (alternateName) this.setLegalName(alternateName)
```

Comment:
This overwrites the actual legal name. I think a better solution is to save the operating name in a new store property, and return it when code asks for legal name (if this is a firm).